contracts/Datastore.sol

```diff
@@ -329,6 +341,147 @@ contract Datastore {
      * @param _entity Entity address     
      */
     function hasWriteAccess(uint _fileId, address _entity) public view returns (bool) {
-        return isOwner(_fileId, _entity) || files[_fileId].permissions[_entity].write;
+        if(isOwner(_fileId, _entity) || files[_fileId].permissions[_entity].write)
+            return true;
+        
+        for (uint i = 0; i < files[_fileId].groupIds.length; i++) {
+            if (files[_fileId].groupPermissions[files[_fileId].groupIds[i]].write) {
+                if (groups[files[_fileId].groupIds[i]].entitiesWithIndex[_entity] != 0)
+                    return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Add a group to the datastore
+     * @param _groupName Name of the group
+     */
+    function createGroup(string _groupName) external {
+        uint id = groupList.length + 1;
+        require(!groups[id].exists);
```

Comment:
In which case `groups[id].exists` would be true?

Reply:
Never. Nice catch!